### PR TITLE
 Orca version checking at startup

### PR DESCRIPTION
**Pull request by mastbaum**
When Orca starts for SNO+, check the current version (in `SNOPModel.h`)
against the current production version, stored in the Orca session
Postgres DB.

The interface to this database (which now handles version checking,
exclusive lock checking, and posting session information) is now
encapsulated within a new class `SessionDB` which the `SNOPModel`
uses as a helper, to help keep the implementation in the `SNOPModel`
under control.

### CODE REVIEW

File: Source/Experiments/SNOP/SessionDB.m [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+            NSLogColor([NSColor redColor], @"Error posting session end to DB: \"%@\"\n", e.reason);
+        }
+    }
+}
+
+
+#pragma mark ���Orca Version
+
+/** Check that the Orca version is up to date. */
+- (void) checkOrcaVersion
```
Comment (mastbaum) [inline Source/Experiments/SNOP/SessionDB.m]: New Orca version checking is here.

**Comment by mastbaum** [comment]
Modal for a version mismatch:
screenshot 2017-11-21 08 29 04

**Comment by tlatorre-uchicago** [comment]
I'm unable to run this on the teststand. Here is the log:

```
17-11-21 15:19:40 -------------------------------------------------
17-11-21 15:19:40    Orca (v11.0h:10332) started in debugger
17-11-21 15:19:40    (After crash or hard debugger stop)           
17-11-21 15:19:40 -------------------------------------------------
17-11-21 15:19:40 Running MacOS Version 10.9.5 (Build 13F1603) 
17-11-21 15:19:40 Mac Address: 10:dd:b1:d1:86:2b
17-11-21 15:19:40 Machine Name: snotdaq2
17-11-21 15:19:40 Launch Path: /Users/snotdaq/orca/DerivedData/Orca/Build/Products/Development/Orca
17-11-21 15:19:40 Orca Command: Ready for connections on port 4
17-11-21 15:19:40 Number of processors: 4
17-11-21 15:19:40 Local Run Control.
17-11-21 15:20:05 Orca Dispatcher: Waiting for connections...
17-11-21 15:20:05 Failed loading: MTC. Reason: *** -[__NSPlaceholderArray initWithObjects:count:]: attempt to insert nil object from objects[0]
17-11-21 15:20:05 Session: hostname: teststand.sp.snolab.ca, orca: 0.12.0, os: Version 10.9.5 (Build 13F1603), key: 54
17-11-21 15:20:06 Opened [~/Desktop/snot_main_teststand.Orca]
17-11-21 15:20:06 Successfully loaded all trigger scans from the database.
17-11-21 15:20:06 Loaded test DB: PMTHV(9728), FEC(2), Crate(1), MTC(1), CAEN(1)
17-11-21 15:20:06 7 MTC settings not loaded!
```

I also get a warning from XCode (not sure if it's relevant):

```
/Users/snotdaq/orca/Source/Experiments/SNOP/SessionDB.m:325:27: warning: Instance method '-terminate:' not found (return type defaults to 'id') [-Wobjc-method-access]
```

**Comment by tlatorre-uchicago** [comment]
Huh, the above error may be unrelated. I tried switching between your branch and master and I was able to run everything fine on master and not on your branch, but now I'm seeing the same error on Ed's branch which is entirely unrelated. I'll try to get to the bottom of it and see what's actually going on here. Hopefully, the Orca file hasn't gotten corrupted.

**Comment by tlatorre-uchicago** [comment]
Ok, seems to be unrelated. I'm unable to reproduce the issue now. No idea what was causing that to happen. If you could just look into the warning reported by XCode then I'll merge.

**Comment by tlatorre-uchicago** [comment]
Also, it looks like the project file changed, so you'll have to merge with master, discard your changes, and then add your files back to the project.

**Comment by mastbaum** [comment]
Ok, will do.

**Comment by mastbaum** [comment]
Merged master, taking the project file from master and re-adding my new files -- hopefully that takes care of it.

**Comment by tlatorre-uchicago** [comment]
merged.

**Pull request closed without merging**